packages/web3-eth/src/utils/watch_transaction_for_confirmations.ts

```diff
@@ -61,4 +56,94 @@ export function watchTransactionForConfirmations<
 			}
 		})() as unknown;
 	}, web3Context.transactionReceiptPollingInterval ?? web3Context.transactionPollingInterval);
+};
+
+const watchBySubscription = ({
+	web3Context,
+	transactionReceipt,
+	transactionPromiEvent,
+	returnFormat,
+}: WaitProps) => {
+	setImmediate(() => {
```

Comment:
This will work for node.js but might have [compatibility](https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate#browser_compatibility) issues in browsers .